In [3]:
import pandas as pd
import mysql.connector
import mysql
import os
from datetime import datetime

host_name = "172.16.0.28" 
user_name = "joaotavares" 
user_password = "@joaotavares"
database='dbfreightech'

def nome_tabela(nome_arquivo, aba):
    tabela = None  
    colunas = None #Initialize tabela to avoid UnboundLocalError 
    retorno = []

    if "INSUMOS" in nome_arquivo:
        if "DESPESAS OPERACIONAIS" == aba:
            tabela = "DESPESAS_OPERACIONAIS_INSUMOS"
            colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `DESCRICAO`, `QUANTIDADE`, `VALOR_TOTAL`, `VALOR_UNITARIO`, `VALOR_POR_EQUIPAMENTO`, `_ID`"
        elif "BENEFÍCIOS REMUNERADOS" == aba:    
            tabela = "BENEFICIO_REMUNERADO_INSUMOS"
            colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `CARGO_BENEFICIO`, `ITEM_BENEFICIO`, `PARTICIPACAO_EMPRESA`, `QUANTIDADE`, `QUANTIDADE_DE_DIAS_UTEIS`, `RESSARCIMENTO_FUNCIONARIO`, `TOTAL`, `TURNO_EMPURRADA`, `VALOR`, `_ID`"
        elif "QLP EQUIPE" == aba: 
            tabela = "QLP_EQUIPE_INSUMOS"
        elif "PARÂMETROS EQUIPE" == aba:
            tabela = "PARAMETROS_EQUIPE_INSUMOS"
        elif "QLP ADM" == aba:
            tabela = "QLP_ADM_INSUMOS"   
        elif "CAVALO" == aba:
            tabela = "CAVALO_INSUMOS"
        elif "CARRETA" == aba:
            tabela = "CARRETA_INSUMOS"
        elif "TRECHOS" == aba:
            tabela = "TRECHOS_INSUMOS"
        elif "CONSUMO" == aba:
            tabela = "CONSUMO_INSUMOS"
        elif "PARÂMETROS MANUTENÇÃO" == aba:
            tabela = "PARAMETROS_MANUTENCAO_INSUMOS"
        elif "PARÂMETROS OPERACIONAIS" == aba:
            tabela = "PARAMETROS_OPERACIONAIS_INSUMOS"
        elif "PNEU" == aba:
            tabela = "PNEU_INSUMOS"
        elif "FATURAMENTO" == aba:
            tabela = "FATURAMENTO_INSUMOS"
        elif "CUSTO EQUIPE" == aba:
            tabela = "CUSTO_EQUIPE_INSUMOS"
        elif "UNIFORMES E EPIS" == aba:
            tabela = "UNIFORMES_EPI_INSUMOS"
        elif "COMBUSTÍVEL" == aba:
            tabela = "COMBUSTIVEL_INSUMOS"     
        elif "ENCARGOS_COM_FERIAS" == aba:
            tabela= "ENCARCOS_COM_FERIAS_INSUMOS"       
        elif"ENCARGOS_SEM_FERIAS" == aba:
            tabela = "ENCARGO_SEM_FERIAS_INSUMOS"       
        elif "PARAMETROS_FISCAL" == aba:   
            tabela = "PARAMETROS_FISCAIS_INSUMOS"                                
        else:
            tabela = aba

    return tabela

def colunas_tabela(tabela):
    colunas = ""  # Initialize colunas to avoid UnboundLocalError home plis 
    if tabela == "DESPESAS_OPERACIONAIS_INSUMOS":
        colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `DESCRICAO`, `QUANTIDADE`, `VALOR_TOTAL`, `VALOR_UNITARIO`, `VALOR_POR_EQUIPAMENTO`, `_ID`"
    elif tabela == "BENEFICIO_REMUNERADO_INSUMOS":
        colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `CARGO_BENEFICIO`, `ITEM_BENEFICIO`, `PARTICIPACAO_EMPRESA`, `QUANTIDADE`, `QUANTIDADE_DE_DIAS_UTEIS`, `RESSARCIMENTO_FUNCIONARIO`, `TOTAL`, `TURNO_EMPURRADA`, `VALOR`, `_ID`"
    elif tabela == "QLP_EQUIPE_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNJP`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `cargoQlpEmpurrada`, `valor0`, `valor15`, `valor30`, `valor50`, `valor60`, `valor70`, `valor80`, `_id`"
    elif tabela == "PARAMETROS_EQUIPE_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `Percentual_Acordo_Coletivo1`, `Percentual_Acordo_Coletivo2`, `Percentual_TO_Remuneracao_Feristas`, `Piso_Salarial_Anterior1`, `Piso_Salarial_Anterior2`, `Ajustado`, `Data_Alteracao_Motorista25`, `Data_Alteracao_Motorista28`, `Data_Alteracao_Motorista40`, `Data_Base`, `Encargo_Provisao_Com_Ferias`, `Encargo_Provisao_Sem_Ferias`, `Fator_Motorista_Produtividade`, `Fator_Motorista_Real`, `Percentual_Acordo_Coletivo_Motorista25`, `Piso_Salarial_Anterior_Motorista25`, `Premio_Produtividade`, `_ID`"
    elif tabela == "QLP_ADM_INSUMOS":
        colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `CARGO`, `DESPESA_BENEFICIO`, `DESPESA_ENCARGOS`, `DESPESA_FROTA_LEVE`, `DESPESA_ORDENADOS`, `DESPESA_TELEFONIA`, `DESPESA_UNIFORMES`, `QLP_BENCHMARK_QUANTIDADE`, `QLP_BENCHMARK_SALARIO`, `QUANTIDADE_BENEFICIO`, `QUANTIDADE_ENCARGOS`, `QUANTIDADE_FROTA_LEVE`, `QUANTIDADE_ORDENADOS`, `QUANTIDADE_TELEFONIA`, `QUANTIDADE_UNIFORMES`, `SALARIO_ENCARGOS`, `SALARIO_ORDENADOS`, `VALE_TRANSPORTE`, `VALOR_BENEFICIO`, `VALOR_FROTA_LEVE`, `VALOR_TELEFONIA`, `VALOR_UNIFORMES`, `_ID`"   
    elif tabela == "CAVALO_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `Custo_Variavel_Simulado`, `Ganhador_BID`, `Padrao`, `Placa`, `Placa_Carreta`, `Spread_BNDES`, `Spread_Banco`, `TJLP`, `Taxa_Finame`, `amortizacaoCavalo`, `ano`, `anoBid`, `ativo`, `cambio`, `carencia`, `chassi`, `ciclo`, `combustivelCapacidade`, `combustivelConsumoBenchmark`, `combustivelConsumoNeg`, `combustivelConsumoNegInteiro`, `combustivelPercentualPerdaVida`, `combustivelVidaCavalo`, `data`, `dataFimContrato`, `eixoEmpurrada`, `faixaKm`, `finameCavalo`, `freeMaintenance`, `ipvaLicenciamento`, `jurosFinameCavalo`, `lucroFixomodeloNovoCicloCavalo`, `lucroVariavelPrevistoCavalo`, `manutencaoBid`, `manutencaoCompraForaDoBidAutorizada`, `manutencaoContrato`, `manutencaoFreeMaintenance`, `manutencaoReaisKm`, `manutencaoReaisKmInteiro`, `manutencaoVidaMeses`, `mesDeEntrada`, `modeloEmpurrada`, `montadora`, `odometroEntrada`, `percentualEntrada`, `percentualIcms`, `percentualReajusteAplicado`, `periodoFiname`, `pneuMedidaEmpurrada`, `reaiskm`, `regiaoInsumos`, `statusFinanciamentoT1Shared`, `telemetriaObrigatoria`, `tipoCombustivelEmpurrada`, `valorIcms`, `valorNfCompra`, `valorPisCofins`, `valorPneu`, `valorReajustado`, `_id`"
    elif tabela == "CARRETA_INSUMOS":
        colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `PLACA`, `SPREAD_BNDES`, `SPREAD_BANCO`, `TJLP`, `TAXA_FINAME`, `AMORTIZACAO_IMPLEMENTO`, `ANO`, `CAPACIDADE_EMPURRADA`, `CAPACIDADE_PALLETS_REAL_EMPURRADA`, `CARENCIA`, `CHASSI`, `CICLO`, `CUSTO_FIXO`, `DATA`, `DATA_FIM_CONTRATO`, `DOUBLE_DECK`, `EIXO_EMPURRADA`, `FAIXA_REFLEXIVA`, `FINAME`, `FINAME_IMPLEMENTO`, `ICMS`, `IMPLEMENTO`, `IPVA_LICENCIAMENTO`, `IPVA_LICENCIAMENTO_MENSAL`, `JUROS_FINAME_IMPLEMENTO`, `LUCRO_FIXOMODELO_NOVO_CICLO`, `LUCRO_FIXOMODELO_NOVO_CICLO_CARRETA`, `LUCRO_VARIAVEL_PREVISTO`, `LUCRO_VARIAVEL_PREVISTO_CARRETA`, `MES_DE_ENTRADA`, `MODELO`, `PERCENTUAL_ENTRADA`, `PERCENTUAL_ICMS`, `PERIODO_FINAME`, `PIS_COFINS`, `PNEU_MEDIDA_EMPURRADA`, `RASTREADOR`, `REVESTIMENTO`, `SEGURO`, `STATUS_FINANCIAMENTO`, `STATUS_FINANCIAMENTO_T1_SHARED`, `TACOGRAFO`, `TIPO_CARROCERIA_EMPURRADA`, `VALOR_ICMS`, `VALOR_NF_COMPRA`, `VALOR_PIS_COFINS`, `VALOR_PNEUS`, `_ID`"
    elif tabela == "TRECHOS_INSUMOS":
        colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `CAPACIDADE`, `DESTINO`, `DESTINO_SAP`, `DESTINO_TMS`, `F_MOV`, `ORIGEM`, `ORIGEM_SAP`, `ORIGEM_TMS`, `CARGA_HORARIA_MOTORISTA_PUXADA_MENSAL`, `CARGA_HORARIA_POR_TRAJETO_MINUTO`, `CARGA_HORARIA_POR_TRAJETO_MINUTO_LUCRO`, `CARGA_HORARIO_TRAJETO_DIA`, `CARGA_HORARIO_TRAJETO_MES`, `CHAVE_TRECHO`, `CNPJ_IDA`, `CNPJ_VOLTA`, `CONSUMO_DIESEL_AJUSTADO`, `CUSTO_DA_DIARIA`, `CUSTO_DO_TR`, `DIAS_MES`, `DIESEL_CONSUMO_DIESEL_REAIS_KM`, `DIESEL_CONSUMO_KM_L`, `FATOR_MOTORISTA_AJUSTADO`, `FATOR_MOTORISTA_INDICADO`, `FATURAMENTO_DESTINO_OBRIGATORIO`, `FRETE_COM_CPRB`, `FRETE_CTRC`, `FRETE_LIQUIDO`, `FRETE_P_TMS`, `FRETE_PIS_COFINS`, `FRETE_REAIS_KM_DIESEL`, `FRETE_REAIS_KM_LAVAGEM`, `FRETE_REAIS_KM_LUCRO_VARIAVEL`, `FRETE_REAIS_KM_MANUTENCAO_CARRETA`, `FRETE_REAIS_KM_MANUTENCAO_CAVALO`, `FRETE_REAIS_KM_PEDAGIO`, `FRETE_REAIS_KM_PNEU`, `FRETE_REAIS_KM_SALARIO_VARIAVEL`, `FRETE_REAIS_KM_SEGURO`, `FRETE_REAIS_VIAGEM_DIESEL`, `FRETE_REAIS_VIAGEM_LAVAGEM`, `FRETE_REAIS_VIAGEM_LUCRO_VARIAVEL`, `FRETE_REAIS_VIAGEM_MANUTENCAO_CARRETA`, `FRETE_REAIS_VIAGEM_MANUTENCAO_CAVALO`, `FRETE_REAIS_VIAGEM_PEDAGIO`, `FRETE_REAIS_VIAGEM_PNEUS`, `FRETE_REAIS_VIAGEM_SALARIO_VARIAVEL`, `FRETE_REAIS_VIAGEM_SEGURO`, `FROTA_NO_MUNICIPIO`, `GRADE_CARREGAMENTO`, `ICMS_ISS`, `IMPOSTOS_ICMS_ISS`, `KM_IDA`, `KM_RODADO`, `KM_RODADO_MES_POR_EQUIPE`, `KM_RODADO_MES_POR_EQUIPE_LUCRO`, `KM_VOLTA`, `LAVAGEM_REAIS_KM`, `LUCRO_VARIAVEL_REAIS_KM`, `MANUTENCAO_CAVALO`, `MANUTENCAO_IMPLEMENTO_REAIS_KM`, `OBSERVACAO`, `PEDAGIO`, `PEDAGIO_POR_EIXO_IDA_VOLTA`, `PEDAGIO_REAIS_KM`, `PERCENTUAL_ICMS_ISS`, `PERCENTUAL_PERDA_DESCARTAVEL`, `PERCENTUAL_PERDA_KM`, `PERCENTUAL_PERDA_REGIAO`, `PNEU_CUSTO_PNEUS_CAMARAS_REAIS_KM`, `PNEU_QUANTIDADE_DE_PNEUS`, `PNEU_VALOR_DE_VENDA_DA_CARCACA`, `PNEU_VALOR_MEDIO_DA_RECAPAGEM`, `PNEU_VALOR_MEDIO_PNEUS`, `PNEU_VIDA_UTIL_PNEU`, `PREMIO_PRODUTIVIDADE_FATOR_MOTORISTA`, `PREMIO_PRODUTIVIDADE_KM_RODADO`, `PREMIO_PRODUTIVIDADE_SALARIO_VARIAVEL`, `PREMIO_PRODUTIVIDADE_SALARIO_VARIAVEL_REAIS_KM`, `PREVISAO_VIAGENS`, `REGIAO_INSUMOS`, `SEGURO`, `SEGURO_REAIS_KM`, `TEMPO_INTERNO_DESTINO`, `TEMPO_INTERNO_DESTINO_LUCRO`, `TEMPO_INTERNO_ORIGEM`, `TEMPO_INTERNO_ORIGEM_LUCRO`, `TEMPO_REFEICAO_MINUTO`, `TEMPO_TRAJETO_FABRICA_CD_MINUTO`, `TRECHO_COM_DIARIA`, `TRECHO_COM_VR`, `TURNO_EMPURRADA`, `TURNOS_FABRICA`, `VELOCIDADE_MEDIA_KM_H`, `VIDA_UTIL_AJUSTADA_PNEU`, `_ID`"
    elif tabela == "CONSUMO_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `capacidadeEmpurrada`, `consumo`, `ford`, `iveco`, `mercedes`, `scania`, `volks`, `volvo`, `_id`"
    elif tabela == "PARAMETROS_MANUTENCAO_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `consumo`, `manutencaoExtra`, `manutencaoReaisKm`, `_id`"
    elif tabela == "PARAMETROS_OPERACIONAIS_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `admConsumoFrotaLeve`, `admCustoFabrica`, `admCustoGasolinaEtanolAnp`, `admCustoGasolinaEtanolComCreditoImpostos`, `admCustoGasolinaEtanolTransportadora`, `admFrotaLeveCarro`, `admRefeitorioFabrica`, `admTelefoniaMovel`, `aliquotaFixaICMSDiesel`, `codigoPromaxOperador`, `codigoPromaxUnidade`, `custoGeralExisteVendaCarcaca`, `custoGeralFaixaReflexiva`, `custoGeralPercentualSeguroCascoTerceiro`, `custoGeralQuantidadeDeRecapagens`, `custoGeralRastreador`, `custoGeralReaisPneuDirecional`, `custoGeralReaisPneuTracao`, `custoGeralRecapagemDirecional`, `custoGeralRecapagemTracao`, `custoGeralRevestimento`, `custoGeralTacografo`, `custoGeralValorVendaCarcaca`, `custoVariavelCustoDiesel`, `custoVariavelCustoDieselComCreditoImpostos`, `custoVariavelCustoPorLavagem`, `custoVariavelFatorDeMotoristaCarro`, `custoVariavelPrecoArlareais`, `custoVariavelPrecoDieselAnp`, `custoVariavelPrecoDieselS10Anp`, `custoVariavelPrecoDieselS10Transportadora`, `custoVariavelPrecoDieselTransportadora`, `custoVariavelQuantidadeDeCarrosEuroIII`, `custoVariavelQuantidadeDeCarrosEuroV`, `custoVariavelQuantidadeDeLavagens`, `faixaBenchmark`, `geralFrotaAtiva`, `geralFrotaInativa`, `geralInicioDataDeVigencia`, `geralInicioOperacao`, `geralKmMedioUnidade`, `geralQuantidadeCarro`, `geralTamanhoOperacaoParaAdm`, `impostoCofinsLucroPresumido`, `impostoCofinsLucroReal`, `impostoContribuicaoPrevSemReceitaBruta`, `impostoCreditoDeIcmsEquipamento`, `impostoCreditoIcmsDiesel`, `impostoCreditoIcmsInsumos`, `impostoCreditoIcmsPneu`, `impostoCreditoIcmsPresumido`, `impostoCreditoPisCofinsInsumo`, `impostoIcms`, `impostoIcmsCavalo`, `impostoIcmsImplemento`, `impostoIcmsParaCustoFixo`, `impostoIss`, `impostoLucroPresumido`, `impostoPercentualDeIcmsParaCreditoDiesel`, `impostoPercentualDeIcmsParaCreditoInsumos`, `impostoPercentualDeIcmsParaCreditoPneu`, `impostoPisLucroPresumido`, `impostoPisLucroReal`, `recIcms`, `recPisCofins`, `regiaoInsumos`, `_id`"
    elif tabela == "PNEU_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `ano`, `pneuMedidaEmpurrada`, `tipoPneu`, `valor`, `_id`"
    elif tabela == "FATURAMENTO_INSUMOS":
        colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `CHECKIMPOSTOICMS`, `CHECKIMPOSTOICMSEQUIPAMENTOEQUIPE`, `CHECKIMPOSTOLUCROPRESUMIDO`, `CHECKIMPOSTOPISCOFINS`, `EQUIPAMENTOATIVOFRETECOMCPRB`, `EQUIPAMENTOATIVOFRETECTRC`, `EQUIPAMENTOATIVOFRETELIQUIDO`, `EQUIPAMENTOATIVOFRETEPISCOFINS`, `EQUIPAMENTOATIVOIMPOSTOSICMSISS`, `EQUIPAMENTOATIVOPERCENTUALICMSISS`, `EQUIPAMENTOATIVOQUANTIDADE`, `EQUIPAMENTOATIVOVALORPARASRTRANS`, `EQUIPAMENTOATIVOVALORUNITARIO`, `EQUIPAMENTOPARADOFRETECOMCPRB`, `EQUIPAMENTOPARADOFRETECTRC`, `EQUIPAMENTOPARADOFRETELIQUIDO`, `EQUIPAMENTOPARADOFRETEPISCOFINS`, `EQUIPAMENTOPARADOIMPOSTOSICMSISS`, `EQUIPAMENTOPARADOPERCENTUALICMSISS`, `EQUIPAMENTOPARADOQUANTIDADE`, `EQUIPAMENTOPARADOVALORPARASRTRANS`, `EQUIPAMENTOPARADOVALORUNITARIO`, `EQUIPEATIVOFRETECOMCPRB`, `EQUIPEATIVOFRETECTRC`, `EQUIPEATIVOFRETELIQUIDO`, `EQUIPEATIVOFRETEPISCOFINS`, `EQUIPEATIVOIMPOSTOSICMSISS`, `EQUIPEATIVOPERCENTUALICMSISS`, `EQUIPEATIVOQUANTIDADE`, `EQUIPEATIVOVALORPARASRTRANS`, `EQUIPEATIVOVALORUNITARIO`, `ESTACIONARIAATIVOFRETECOMCPRB`, `ESTACIONARIAATIVOFRETECTRC`, `ESTACIONARIAATIVOFRETELIQUIDO`, `ESTACIONARIAATIVOFRETEPISCOFINS`, `ESTACIONARIAATIVOIMPOSTOSICMSISS`, `ESTACIONARIAATIVOPERCENTUALICMSISS`, `ESTACIONARIAATIVOQUANTIDADE`, `ESTACIONARIAATIVOVALORPARASRTRANS`, `ESTACIONARIAATIVOVALORUNITARIO`, `OUTRADESPESAFRETECOMCPRB`, `OUTRADESPESAFRETECTRC`, `OUTRADESPESAFRETELIQUIDO`, `OUTRADESPESAFRETEPISCOFINS`, `OUTRADESPESAIMPOSTOSICMSISS`, `OUTRADESPESAPERCENTUALICMSISS`, `OUTRADESPESAQUANTIDADE`, `OUTRADESPESAVALORPARASRTRANS`, `OUTRADESPESAVALORUNITARIO`, `QLPADM_FRETECOMCPRB`, `QLPADM_FRETECTRC`, `QLPADM_FRETELIQUIDO`, `QLPADM_FRETEPISCOFINS`, `QLPADM_IMPOSTOSICMSISS`, `QLPADM_PERCENTUALICMSISS`, `QLPADM_QUANTIDADE`, `QLPADM_VALORPARASRTRANS`, `QLPADM_VALORUNITARIO`, `_ID`"
    elif tabela =="CUSTO_EQUIPE_INSUMOS":
        colunas = "`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `ASSISTENCIA_MEDICA`, `BENEFICIO_FIXO`, `CAFE_DA_MANHA`, `CARGO_EQUIPE_EMPURRADA`, `CESTA_BASICA`, `DIARIA`, `ENCARGO_PROVISAO`, `FATOR_MOTORISTA`, `FROTA_ATIVA`, `OUTRO`, `PCMSO_POR_MES`, `PLR`, `SALARIO_FIXO`, `SEGURO_DE_VIDA`, `TICKET_REFEICAO_LIQUIDO`, `TOTAL`, `TOTAL_UNIFORME_EPI`, `VALE_TRANSPORTE_LIQUIDO`, `_ID`"    
    elif tabela =="UNIFORMES_EPI_INSUMOS":
        colunas="`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `CUSTO`, `CUSTO_MES`, `ICMS_REGIAO`, `ITEM_EPI`, `QUANTIDADE_ANO`, `TIPO_EQUIPE`, `_ID`"
    elif tabela =="COMBUSTIVEL_INSUMOS":
        colunas="`VIGENCIA`, `UNIDADE_CNPJ`, `UNIDADE_NOME`, `UNIDADE_SAP`, `UNIDADE_TMS`, `UNIDADE_PROMAX_UNB`, `UNIDADE_REGIONAL`, `OPERADOR_CNPJ`, `OPERADOR_NOME`, `OPERADOR_SAP`, `OPERADOR_TMS`, `OPERADOR_PROMAX`, `ORGANIZACAO_DE_COMPRAS`, `PRAZO_PAGAMENTO`, `CREDITOIMPOSTO`, `DESCRICAO`, `INICIATIVA`, `PRECOANP`, `PRECOCREDITOIMPOSTOS`, `PRECOOPERADORA`, `_ID`"
    elif tabela =="ENCARCOS_COM_FERIAS_INSUMOS":
        colunas ="`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `auxilioDoenca`, `auxilioDoencaRpsA`, `auxilioDoencaRpsB`, `auxilioDoencaRpsC`, `avisoPrevio`, `avisoPrevioRpsA`, `avisoPrevioRpsB`, `avisoPrevioRpsC`, `decimoTerceiroSalario`, `decimoTerceiroSalarioRpsA`, `decimoTerceiroSalarioRpsB`, `decimoTerceiroSalarioRpsC`, `diasDeAviso`, `diferencaProvisaoFerias`, `diferencaProvisaoFeriasRpsA`, `diferencaProvisaoFeriasRpsB`, `diferencaProvisaoFeriasRpsC`, `feriasNormais`, `feriasNormaisRpsA`, `feriasNormaisRpsB`, `feriasNormaisRpsC`, `feriasProporcionaisIndenizadas`, `feriasProporcionaisIndenizadasFormula`, `feriasProporcionaisIndenizadasRpsA`, `feriasProporcionaisIndenizadasRpsB`, `feriasProporcionaisIndenizadasRpsC`, `feriasUmTerco`, `feriasUmTercoRpsA`, `feriasUmTercoRpsB`, `feriasUmTercoRpsC`, `feristaFerias`, `feristaFeriasRpsA`, `feristaFeriasRpsB`, `feristaFeriasRpsC`, `fgts`, `incidenciaCumulativa`, `incidenciaCumulativaFerista`, `incidenciaCumulativaFeristaRpsA`, `incidenciaCumulativaFeristaRpsB`, `incidenciaCumulativaFeristaRpsC`, `incidenciaEngargosFerista`, `incidenciaEngargosFeristaRpsA`, `incidenciaEngargosFeristaRpsB`, `incidenciaEngargosFeristaRpsC`, `incidenciaUmTercoFeriasSobreAviso`, `incidenciaUmTercoFeriasSobreAvisoRpsA`, `incidenciaUmTercoFeriasSobreAvisoRpsB`, `incidenciaUmTercoFeriasSobreAvisoRpsC`, `incra`, `inss`, `involuntario`, `mesDissidio`, `multaFGTS`, `salarioEducacao`, `sebrae`, `seguroAcidenteTrabalho`, `senat`, `sest`, `tempoMedioCasa`, `total`, `totalEncargos`, `totalFerista`, `totalProvisoes`, `totalRpsA`, `totalRpsB`, `totalRpsC`, `turnOver`, `voluntario`, `_id`"
    elif tabela =="ENCARGO_SEM_FERIAS_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `auxilioDoenca`, `auxilioDoencaRpsA`, `auxilioDoencaRpsB`, `auxilioDoencaRpsC`, `avisoPrevio`, `avisoPrevioRpsA`, `avisoPrevioRpsB`, `avisoPrevioRpsC`, `decimoTerceiroSalario`, `decimoTerceiroSalarioRpsA`, `decimoTerceiroSalarioRpsB`, `decimoTerceiroSalarioRpsC`, `diasDeAviso`, `diferencaProvisaoFerias`, `diferencaProvisaoFeriasRpsA`, `diferencaProvisaoFeriasRpsB`, `diferencaProvisaoFeriasRpsC`, `feriasNormais`, `feriasNormaisRpsA`, `feriasNormaisRpsB`, `feriasNormaisRpsC`, `feriasProporcionaisIndenizadas`, `feriasProporcionaisIndenizadasFormula`, `feriasProporcionaisIndenizadasRpsA`, `feriasProporcionaisIndenizadasRpsB`, `feriasProporcionaisIndenizadasRpsC`, `feriasUmTerco`, `feriasUmTercoRpsA`, `feriasUmTercoRpsB`, `feriasUmTercoRpsC`, `feristaFerias`, `feristaFeriasRpsA`, `feristaFeriasRpsB`, `feristaFeriasRpsC`, `fgts`, `incidenciaCumulativa`, `incidenciaCumulativaFerista`, `incidenciaCumulativaFeristaRpsA`, `incidenciaCumulativaFeristaRpsB`, `incidenciaCumulativaFeristaRpsC`, `incidenciaEngargosFerista`, `incidenciaEngargosFeristaRpsA`, `incidenciaEngargosFeristaRpsB`, `incidenciaEngargosFeristaRpsC`, `incidenciaUmTercoFeriasSobreAviso`, `incidenciaUmTercoFeriasSobreAvisoRpsA`, `incidenciaUmTercoFeriasSobreAvisoRpsB`, `incidenciaUmTercoFeriasSobreAvisoRpsC`, `incra`, `inss`, `involuntario`, `mesDissidio`, `multaFGTS`, `salarioEducacao`, `sebrae`, `seguroAcidenteTrabalho`, `senat`, `sest`, `tempoMedioCasa`, `total`, `totalEncargos`, `totalFerista`, `totalProvisoes`, `totalRpsA`, `totalRpsB`, `totalRpsC`, `turnOver`, `voluntario`, `_id`"
    elif tabela == "PARAMETROS_FISCAIS_INSUMOS":
        colunas = "`Vigencia`, `Unidade_CNPJ`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `ICMS`, `ISS`, `cdd`, `creditoIcms`, `ded1`, `ded2`, `geo`, `pisCofinsDed1`, `pisCofinsDed2`, `pisCofinsSpot1`, `pw`, `sap`, `sim`, `spot1`, `uf`, `_id`"


    return colunas

    
def conectar_banco(host_name, user_name, user_password, database):
    try:
        connection = mysql.connector.connect(
            host=host_name,
            user=user_name,
            password=user_password,
            database=database
        )
        # print ("Conexão ao banco de dados MySQL realizada com sucesso")
    except mysql.connector.Error as err:
        print(f"Erro ao conectar ao banco de dados: '{err}'")
    return connection        

def encontrar_arquivo_mais_recente(diretorio, prefixo):
    arquivos = [f for f in os.listdir(diretorio) if f.startswith(prefixo) and f.endswith('.xlsx')]
    if not arquivos:
        return None
    return max(arquivos, key=lambda f: os.path.getmtime(os.path.join(diretorio, f)))

diretorio =  r'C:\\Freightech\\load\\INSUMOS'
prefixo = 'INSUMOS'

nome_arquivo_mais_recente = encontrar_arquivo_mais_recente(diretorio, prefixo)

if nome_arquivo_mais_recente:
    nome_arquivo = os.path.join(diretorio, nome_arquivo_mais_recente)
    print(f"Arquivo mais recente encontrado: {nome_arquivo}")
    
    arquivo_excel = pd.ExcelFile(nome_arquivo)
    nomes_abas = arquivo_excel.sheet_names

    for nome_aba in nomes_abas:
        arquivo = pd.read_excel(nome_arquivo, sheet_name=nome_aba)
        tabelaInsert = nome_tabela(nome_arquivo, nome_aba)
        colunasInsert = colunas_tabela(tabelaInsert)
        colunas = arquivo.columns.tolist()
        todosValores = ""
        for index, linha in arquivo.iterrows():
            valores = linha.tolist()
            valores_string = ','.join(['"'+str(valor).rstrip('\'"')+'"' for valor in valores])
            valores_string = "(" + valores_string + "),"
            todosValores += valores_string


        todosValores = todosValores.rstrip(',')
        if not colunasInsert:
            print("Erro: colunasInsert está vazio. Encerrando a aplicação.")
            exit(1)
        sql = f"INSERT IGNORE INTO {tabelaInsert}({colunasInsert}) VALUES {todosValores}"


        # print(sql)
        conexao = conectar_banco(host_name, user_name, user_password, database)
        cursor = conexao.cursor()
        try:
            cursor.execute(sql)
            conexao.commit()
            print(f"Query executada com sucesso na tabela {nome_aba}.")
        except Exception as e:
            conexao.rollback()
            print(f"Erro ao executar a query: {e}")
            print(sql) 
            
        finally:
            cursor.close()
            conexao.close()
else:
    print("Nenhum arquivo encontrado com o prefixo especificado.")


Arquivo mais recente encontrado: C:\\Freightech\\load\\INSUMOS\INSUMOS (6).xlsx


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela DESPESAS OPERACIONAIS.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela BENEFÍCIOS REMUNERADOS.
Erro ao executar a query: 1136 (21S01): Column count doesn't match value count at row 1
INSERT IGNORE INTO QLP_EQUIPE_INSUMOS(`Vigencia`, `Unidade_CNJP`, `Unidade_Nome`, `Unidade_SAP`, `Unidade_TMS`, `Unidade_Promax_UNB`, `Unidade_Regional`, `Operador_CNPJ`, `Operador_Nome`, `Operador_SAP`, `Operador_TMS`, `Operador_Promax`, `Organizacao_de_Compras`, `Prazo_Pagamento`, `cargoQlpEmpurrada`, `valor0`, `valor15`, `valor30`, `valor50`, `valor60`, `valor70`, `valor80`, `_id`) VALUES ("INSUMOS_2_1_2025","07526557001505_LT","LATA CAMAÇARI","BR04","nan","nan","nan","20618821000799","OPERALOG","390268","nan","1","nan","nan","105.14","385.19","0","Cargo: MOTORISTA 28","120.0","2195.07","21.03","2000-01-01 00:00:00","0","40.12","51.32","2643.61","32.87","2537.87","4.0","0","5307.65","4191.16","3840.23","2769.78","5.25","55.65","7341.99","3080.04","10695.43","1421.38","8031.4","70.79","Cargo: EQUIPE ATIVA 8x16","245.88","0","0","1534759

C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")
C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela PARÂMETROS EQUIPE.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela QLP ADM.
Query executada com sucesso na tabela CAVALO.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")
C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela CARRETA.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela TRECHOS.
Query executada com sucesso na tabela CONSUMO.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")
C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela PARÂMETROS MANUTENÇÃO.
Query executada com sucesso na tabela PARÂMETROS OPERACIONAIS.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")
C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela PNEU.
Query executada com sucesso na tabela FATURAMENTO.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")
C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela CUSTO EQUIPE.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela UNIFORMES E EPIS.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela COMBUSTÍVEL.
Query executada com sucesso na tabela ENCARGOS_COM_FERIAS.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")
C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela ENCARGOS_SEM_FERIAS.


C:\Users\ADMIN.JOAO\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\openpyxl\styles\stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Query executada com sucesso na tabela PARAMETROS_FISCAL.


        "DESPESAS_OPERACIONAIS_INSUMOS",-490-588 
        "BENEFICIO_REMUNERADO_INSUMOS",-3460-3460  
        "QLP_EQUIPE_INSUMOS",-408-12 
        "PARAMETROS_EQUIPE_INSUMOS",-12 -12 
        "QLP_ADM_INSUMOS",-12 - 408 
        "CAVALO_INSUMOS",-195-195 
        "CARRETA_INSUMOS",-255 - 255 
        "TRECHOS_INSUMOS",-246 -246 
        "CONSUMO_INSUMOS",-92 - 92 
        "PARAMETROS_MANUTENCAO_INSUMOS",-12 -12 
        "PARAMETROS_OPERACIONAIS_INSUMOS",-12 -12 
        "PNEU_INSUMOS",-192 -192 
        "FATURAMENTO_INSUMOS",-18-18 
        "CUSTO_EQUIPE_INSUMOS",-36 -36 
        "UNIFORMES_EPIS_INSUMOS",-336 - 336 
        "COMBUSTIVEL_INSUMOS"-24- 24 
        "ENCARCO_COM_FERIAS_INSUMOS"-6-
        "ENCARCO_SEM_FERIAS_INSUMOS"-6- 
        


In [3]:
import sys
print(sys.executable)

C:\Users\ADMIN.JOAO\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\python.exe
